File: lc121.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        result = 0

        for i in range(len(prices) - 1):
            if prices[i] < prices[i + 1]:
                if prices[i + 1] - prices[i] > result:
                    result = prices[i + 1] - prices[i]
                    
                for j in range(i + 1, len(prices)):
                    if prices[j] > prices[i]:
                        if prices[j] - prices[i] > result:
                            result = prices[j] - prices[i]
                        else:
                            continue
                else:
                    continue
            else:
                continue
        return result

            

File: lc121.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices == null || prices.length == 0) return 0;
        
        int minBuy = prices[0];
        int maxProfit = 0;
        
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] < minBuy) {
                minBuy = prices[i];
            } else {
               maxProfit = Math.max(maxProfit, prices[i] - minBuy); 
            }
        }
        return maxProfit;
    }
}